cmd/thanos/receive.go

```diff
@@ -277,6 +290,8 @@ func runReceive(
 			}
 			s := newStoreGRPCServer(logger, reg, tracer, tsdbStore, opts)
 
+			// Wait hashring to be ready before start serving metrics
+			<-hashringReady
```

Comment:
Why are we waiting for the hashring to be ready before serving metrics from the store? These things are entirely independent IMO